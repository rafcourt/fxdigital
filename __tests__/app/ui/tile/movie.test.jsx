import '@testing-library/jest-dom'
import { render, screen, fireEvent } from '@testing-library/react'
import MovieTile from '@/app/ui/tile/movie'
import mockRouter from 'next-router-mock'
import renderer from 'react-test-renderer';

jest.mock('next/navigation', () => jest.requireActual('next-router-mock'));

describe('Movie Tile', () => {
    it('Renders a tile containing a link and an image.', () => {
        mockRouter.push("/home");
        render(<MovieTile />);
    
        const image = screen.getByRole('img');
        const link = screen.getByRole('link');
    
        expect(image).toBeInTheDocument();
        expect(link).toBeInTheDocument();
    })
    it('Matches the src of the image and the href of the link in the tile.', () => {
        mockRouter.push("/home");
        const mockTile = {
            original_title: 'Blockbuster Movie',
            poster_path: '/mock/poster/path.png',
            id: 123456,
            media_type: 'movie'
        };
        render(<MovieTile 
            title={ mockTile.original_title } 
            imagePath={ mockTile.poster_path }
            id={ mockTile.id }
            media_type={ mockTile.media_type }
            />);
    
        const image = screen.getByRole('img');
        const link = screen.getByRole('link');
    
        expect(image.src).toMatch(new RegExp(encodeURIComponent(`https://image.tmdb.org/t/p/w185${mockTile.poster_path}`)));
        expect(link.href).toMatch(new RegExp(`/details/${mockTile.media_type}/${mockTile.id}`));
    })
    it.skip('Opens the details screen', () => {
        mockRouter.push("/home");
        const mockTile = {
            id: 123456,
            media_type: 'movie'
        };
        render(<MovieTile 
            id={ mockTile.id }
            media_type={ mockTile.media_type }
            />);

        const link = screen.getByTestId ('testid');
        fireEvent.keyPress(link, {key: 'Enter', code: 'Enter', charCode: 13});
        
        expect(mockRouter).toMatchObject({ 
            asPath: `/details/${mockTile.media_type}/${mockTile.id}`,
          });
    })

    it('snapshot matches', () => {
        const mockTile = {
            original_title: 'Blockbuster Movie',
            poster_path: '/mock/poster/path.png',
            id: 123456,
            media_type: 'movie'
        };
        const tree = renderer
          .create(<MovieTile 
            title={ mockTile.original_title } 
            imagePath={ mockTile.poster_path }
            id={ mockTile.id }
            media_type={ mockTile.media_type }
            testId="test-id"
            />)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
  })

