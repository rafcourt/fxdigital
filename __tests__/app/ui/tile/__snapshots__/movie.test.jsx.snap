// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Tile snapshot matches 1`] = `
<div
  className=" w-movie-tile h-fit inline-block relative text-center px-1 py-1 mx-2 rounded-l bg-transparent"
  data-testid="test-id"
>
  <a
    href="/details/movie/123456"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    <img
      alt="Blockbuster Movie"
      className="rounded-l bg-gray-500"
      data-nimg="1"
      decoding="async"
      height={213}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2Fmock%2Fposter%2Fpath.png&w=384&q=75"
      srcSet="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2Fmock%2Fposter%2Fpath.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw185%2Fmock%2Fposter%2Fpath.png&w=384&q=75 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={150}
    />
  </a>
</div>
`;
